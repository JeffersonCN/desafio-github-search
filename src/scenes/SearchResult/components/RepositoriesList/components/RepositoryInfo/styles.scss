.repositoryInfo {
    font-family: 'Raleway';
    padding: 0 0 30px 44px;
    max-width: 750px;

    .linkToRepo {
        font-size: 35px;
        font-weight: 800;
        color: #ac53f2;
        text-decoration: none;

        &:hover {
            text-decoration: underline;
        }
    }

    .repositoryName {
        margin: 0;
    }

    .repositoryDescription {
        font-size: 20px;
        font-weight: 300;
        margin: 0 0 12px;
    }
}