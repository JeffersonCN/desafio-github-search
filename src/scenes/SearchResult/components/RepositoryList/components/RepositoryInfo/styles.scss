.repository-info {
    font-family: 'Raleway';
    padding: 15px;

    .repository-name {
        font-size: 35px;
        font-weight: 800;
        color: var(--primary-color);
        margin: 0;
    }

    .repository-description {
        font-size: 20px;
        font-weight: 300;
        margin: 0 0 15px;
    }
}