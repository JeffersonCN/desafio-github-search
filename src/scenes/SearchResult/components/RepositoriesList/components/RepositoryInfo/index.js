import React, { Component } from 'react';
import TextWithLeftIcon from  '../../../../../../components/TextWithLeftIcon';
import starIcon from '../../../../assets/icons/star-icon.svg';

import styles from './styles.scss';

class RepositoryInfo extends Component {
    render() {
        return (
            <div className={styles.repositoryInfo}>
                <p className={styles.repositoryName}>
                    <a className={styles.linkToRepo} target="_blank" href={this.props.repository.html_url}>{this.props.repository.name}</a>
                </p>
                <p className={styles.repositoryDescription}>{this.props.repository.description}</p>
                <TextWithLeftIcon icon={starIcon} text={this.props.repository.stargazers_count} />
            </div>
        );
    }
}

export default RepositoryInfo;