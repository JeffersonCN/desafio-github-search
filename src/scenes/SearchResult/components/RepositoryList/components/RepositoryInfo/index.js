import React, { Component } from 'react';
import TextWithLeftIcon from  '../TextWithLeftIcon';
import starIcon from '../../../../../../assets/icons/star-icon.svg';

import './styles.scss';

class RepositoryInfo extends Component {
    render() {
        return (
            <div className="repository-info">
                <p className="repository-name">{this.props.repository.name}</p>
                <p className="repository-description">{this.props.repository.description}</p>
                <TextWithLeftIcon icon={starIcon} text={this.props.repository.stargazers_count} />
            </div>
        );
    }
}

export default RepositoryInfo;