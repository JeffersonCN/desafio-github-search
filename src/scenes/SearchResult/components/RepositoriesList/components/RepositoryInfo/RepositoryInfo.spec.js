import React from 'react';
import { shallow } from 'enzyme';
import RepositoryInfo from './';

describe('Testing RepositoryInfo Component', () => {
    it('shoud render correctly', () => {
        const repo = {
            name: "Repositório",
            description: "Descrição",
            stargazers_count: 10,
            html_url: "https://github.com/JeffersonCN/Balan-o-Patrimonial"
        }

        const wrapper = shallow(
            <RepositoryInfo repository={repo}/>
        );

        expect(wrapper).toMatchSnapshot();
    });
});