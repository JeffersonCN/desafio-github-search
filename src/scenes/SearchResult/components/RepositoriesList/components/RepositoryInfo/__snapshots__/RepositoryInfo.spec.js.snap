// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RepositoryInfo Component shoud render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RepositoryInfo
    repository={
        Object {
            "description": "Descrição",
            "html_url": "https://github.com/JeffersonCN/Balan-o-Patrimonial",
            "name": "Repositório",
            "stargazers_count": 10,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className={undefined}
>
          <a
                    className={undefined}
                    href="https://github.com/JeffersonCN/Balan-o-Patrimonial"
                    target="_blank"
          >
                    Repositório
          </a>
</p>,
        <p
          className={undefined}
>
          Descrição
</p>,
        <TextWithLeftIcon
          icon="star-icon.svg"
          text={10}
/>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className={undefined}
            href="https://github.com/JeffersonCN/Balan-o-Patrimonial"
            target="_blank"
>
            Repositório
</a>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Repositório",
            "className": undefined,
            "href": "https://github.com/JeffersonCN/Balan-o-Patrimonial",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "Repositório",
          "type": "a",
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Descrição",
          "className": undefined,
        },
        "ref": null,
        "rendered": "Descrição",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "icon": "star-icon.svg",
          "text": 10,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className={undefined}
>
            <a
                        className={undefined}
                        href="https://github.com/JeffersonCN/Balan-o-Patrimonial"
                        target="_blank"
            >
                        Repositório
            </a>
</p>,
          <p
            className={undefined}
>
            Descrição
</p>,
          <TextWithLeftIcon
            icon="star-icon.svg"
            text={10}
/>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className={undefined}
              href="https://github.com/JeffersonCN/Balan-o-Patrimonial"
              target="_blank"
>
              Repositório
</a>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Repositório",
              "className": undefined,
              "href": "https://github.com/JeffersonCN/Balan-o-Patrimonial",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Repositório",
            "type": "a",
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Descrição",
            "className": undefined,
          },
          "ref": null,
          "rendered": "Descrição",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "icon": "star-icon.svg",
            "text": 10,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
